src/software/ai/hl/stp/play/play.cpp

```diff
@@ -39,6 +39,31 @@ std::vector<std::shared_ptr<Tactic>> Play::getTactics(const World &world)
     return std::vector<std::shared_ptr<Tactic>>();
 }
 
+std::vector<std::unique_ptr<Intent>> Play::get(
+    RobotToTacticAssignmentFunction robot_to_tactic_assignment_algorithm,
+    MotionConstraintBuildFunction motion_constraint_builder, const World &new_world)
+{
+    std::vector<std::unique_ptr<Intent>> intents;
+    auto tactics = getTactics(new_world);
+    std::vector<std::shared_ptr<const Tactic>> const_tactics;
+    const_tactics.reserve(tactics.size());
+    std::transform(tactics.begin(), tactics.end(), std::back_inserter(const_tactics),
+                   [](std::shared_ptr<Tactic> tactic) { return tactic; });
+    auto robot_tactic_assignment =
+        robot_to_tactic_assignment_algorithm(const_tactics, new_world);
+    for (auto tactic : tactics)
+    {
+        auto iter = robot_tactic_assignment.find(tactic);
+        if (iter != robot_tactic_assignment.end())
+        {
+            auto intent = tactic->get(iter->second, new_world);
+            intent->setMotionConstraints(motion_constraint_builder(*tactic));
+            intents.push_back(std::move(intent));
+        }
```

Comment:
We should probably log at least a warning if the robot doesn't exist.

Reply:
What if, god forbid, we get a yellow card and we have 5 robots vs 6 tactics? Or we're field testing. I think it's too easy to get into situations where this warning will spam our logs.